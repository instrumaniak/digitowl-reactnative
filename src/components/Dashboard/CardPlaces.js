/**
 *   Dashboard: Places section
 */

import React, { PureComponent } from 'react'
import { View, StyleSheet } from 'react-native'

import {
  Card,
  CardItem,
  Text
} from 'native-base'

import { colors } from '../../config'
import CategoryItem from './CategoryItem'


class CardPlaces extends PureComponent {
  render() {
    return (
      <Card style={{ elevation: 2 }}>
        <CardItem>
          <Text style={styles.cardHeaderText}>Places</Text>
        </CardItem>

      {/* Category Item list */}
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Office"
            description="26 transations"
            value="-1 804.90 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Home"
            description="17 transactions"
            value="-694.20 USD"
          />
        </CardItem>
        <CardItem>
          <CategoryItem
            iconText="$"
            title="Cafe"
            description="1 transactions"
            value="-155.40 USD"
          />
        </CardItem>

        <CardItem>
          <View style={{
            flex: 1,
            alignItems: 'center'
          }}>
            <Text style={{
              color: colors.grey1,
              textDecorationLine: 'underline'
            }}>All places (9)</Text>
          </View>
        </CardItem>
      </Card>
    )
  }
}

const styles = StyleSheet.create({
  cardHeaderText: {
    color: colors.grey1
  },
  btnContainer: {
    flex: 1,
    flexDirection: 'row'
  },
  btnLight: {
    backgroundColor: 'white',
    flex: 1,
    marginHorizontal: 5,
    justifyContent: 'center'
  },
  btnText: {
    //color: colors.greenDark
  },
  overviewButton: {
    flex: 1,
    alignItems: 'center',
    padding: 5
  },
  oBtnActive: {
    backgroundColor: 'rgba(0,0,0,0.05)',
    borderRadius: 5
  },
  obtnHeading: {
    color: colors.grey1,
    marginBottom: 5,
    fontSize: 15
  },
  obtnValue: {
    color: colors.grey1,
    fontSize: 20,
    fontWeight: 'bold'
  }
})

export default CardPlaces
