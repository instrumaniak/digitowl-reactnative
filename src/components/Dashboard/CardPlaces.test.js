/**
 *   Tests for component: CardPlaces
 */

import React from 'react'
import CardPlaces from './CardPlaces'
import renderer from 'react-test-renderer'

describe('CardPlaces component', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<CardPlaces />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
