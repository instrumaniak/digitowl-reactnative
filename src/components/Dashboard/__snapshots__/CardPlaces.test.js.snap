// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPlaces component Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "elevation": 2,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#5B6267",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "600",
            "letterSpacing": -0.408,
            "lineHeight": 22,
          },
        ]
      }
      uppercase={false}
    >
      Places
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#c3c6c7",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 28,
              "fontWeight": "400",
              "letterSpacing": 0.355469,
              "lineHeight": 34,
            }
          }
        >
          $
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#5B6267",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "400",
                "letterSpacing": -0.408,
                "lineHeight": 22,
              }
            }
          >
            Office
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#c3c6c7",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "letterSpacing": -0.24,
                "lineHeight": 20,
              }
            }
          >
            26 transations
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#4B5358",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "letterSpacing": -0.408,
                "lineHeight": 22,
              }
            }
          >
            -1 804.90 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#c3c6c7",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 28,
              "fontWeight": "400",
              "letterSpacing": 0.355469,
              "lineHeight": 34,
            }
          }
        >
          $
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#5B6267",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "400",
                "letterSpacing": -0.408,
                "lineHeight": 22,
              }
            }
          >
            Home
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#c3c6c7",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "letterSpacing": -0.24,
                "lineHeight": 20,
              }
            }
          >
            17 transactions
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#4B5358",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "letterSpacing": -0.408,
                "lineHeight": 22,
              }
            }
          >
            -694.20 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#c3c6c7",
            "borderRadius": 25,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 28,
              "fontWeight": "400",
              "letterSpacing": 0.355469,
              "lineHeight": 34,
            }
          }
        >
          $
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#5B6267",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "400",
                "letterSpacing": -0.408,
                "lineHeight": 22,
              }
            }
          >
            Cafe
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#c3c6c7",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "letterSpacing": -0.24,
                "lineHeight": 20,
              }
            }
          >
            1 transactions
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#4B5358",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "letterSpacing": -0.408,
                "lineHeight": 22,
              }
            }
          >
            -155.40 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "backgroundColor": "transparent",
              "color": "#5B6267",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "letterSpacing": -0.24,
              "lineHeight": 20,
              "textDecorationLine": "underline",
            },
          ]
        }
        uppercase={false}
      >
        All places (9)
      </Text>
    </View>
  </View>
</View>
`;
